uk.co.compucorp.civicrm.hrleaveandabsences/js/src/leave-absences/leave-type-wizard/form/components/leave-type-wizard-form.component.js

```diff
@@ -12,7 +14,165 @@ define([], function () {
+              lineBreak,
+              'If you enable this option the leave type will simply be given the title "Leave".',
+              'Managers and administrators will continue to see the actual leave title of the request.',
+              lineBreak,
+              'This may be helpful if the leave type title is sensitive - i.e. Unpaid leave or Compassionate leave.'
+            ].join('')
+          }
+        ]
+      },
+      {
+        name: 'leave-requests',
+        label: 'Leave Requests',
+        fields: [
+          {
+            name: 'max_consecutive_leave_days',
+            label: 'Max consecutive duration (Leave blank for unlimited)',
+            helpText: [
+              'Configure the maximum duration of consecutive leave permitted to be selected in a single leave request.',
+              'You can leave this field blank for unlimited duration of leave.'
+            ].join('')
+          }
+        ]
+      },
+      {
+        name: 'public-holidays',
+        label: 'Public Holidays',
+        fields: [
+          {
+            name: 'must_take_public_holiday_as_leave',
+            label: 'Do staff work on public holidays?'
+          }
+        ]
+      },
+      {
+        name: 'carry-forwards',
+        label: 'Carry Forwards',
+        fields: [
+          {
+            name: 'allow_carry_forward',
+            label: 'Allow carry forward?'
+          }
+        ]
+      }
+    ];
+
+    vm.$onInit = $onInit;
+    vm.getSettingsTabFields = getSettingsTabFields;
+    vm.openSection = openSection;
+    vm.openSettingsTab = openSettingsTab;
+    vm.selectLeaveTypeCategory = selectLeaveTypeCategory;
+
+    function $onInit () {
+      initDefaultView();
+    }
+
+    /**
+     * Gets fields for the currently active settings tab
+     *
+     * @return {Array} collection of fields
+     */
+    function getSettingsTabFields () {
+      var activeTab = _.find(vm.settingsTabs, { active: true });
+
+      return activeTab.fields;
+    }
+
+    /**
+     * Initiates the default view:
+     * - selects Leave category;
+     * - expands the General section and leaves the Settings section collapsed;
+     * - selects Basic Details settings tab.
+     */
+    function initDefaultView () {
+      vm.leaveTypeCategory = 'leave';
+
+      openSection('general');
+      openSettingsTab('basic-details');
+    }
+
+    /**
+     * Opens a section by its name and collapses all other sections
+     *
+     * @param {String} sectionName
+     */
+    function openSection (sectionName) {
+      vm.sections.forEach(function (section) {
+        section.active = false;
+      });
+
+      _.find(vm.sections, { name: sectionName }).active = true;
+    }
+
+    /**
+     * Opens a settings tab by its name and collapses all other settings tabs
+     *
+     * @param {String} settingsTabName
+     */
+    function openSettingsTab (settingsTabName) {
```

Comment:
could you make the function more generic without referencing the "settings" section at all? It would make the code more flexible. Something like
```js
function openSectionTab(sectionName, tabName) {

}
```
this would also get rid of all the "settings" hardcoded references of `vm.settingsTab` and `vm.activeSettingsTabIndex` etc

Reply:
I do not think we need that level of flexibility now. If we want to adapt the wizard for every future forms, we will need to compare and analyse all other planned wizards.

Reply:
I attempted to do what you asked.